### PR TITLE
fix(RadioGroup): use regular components instead of Label

### LINKED ISSUE #1040: RadioGroupItem shouldn't use a Stardust Label component
## Bug Report
The RadioGroupItem shouldn't use a Stardust Label component, it should use an html <label> for it's items.

Do we need an input for accessibility?

```
<input name="aaa" type="radio" />
<label for="aaa">

</label>
```

### PR DESCRIPTION
**Pull request by layershifter**
Fixes #1040.

---

# BREAKING CHANGES

1. Generated markup was changed:

### Before

```html
    <div
      role="radio"
      tabindex="-1"
      aria-checked="false"
      aria-disabled="true"
      class="ui-radiogroup__item dd aj"
    >
      <span class="ui-label bt ht et gy hz ii cw ib ic id bf bm hu">
        <span class="ui-icon ek aj bu ak ai ie if ig ce hv ij ik hy" aria-hidden="true"></span>
        <span dir="auto">Prosciutto</span>
      </span>
    </div>
```

### After
```html
    <div
      role="radio"
      tabindex="-1"
      aria-checked="false"
      aria-disabled="true"
      class="ui-radiogroup__item jo bf bt bm jp dd jq ie"
    >
      <span class="ui-icon ek aj bu ak ai ju jv jw ce jr il jy jt" aria-hidden="true"></span>
      <span dir="auto" class="ui-box">Prosciutto</span>
    </div>
```

2. `label` is now shorthand value, see [Shorthand Props](https://stardust-ui.github.io/react/shorthand-props)

### CODE REVIEW

File: packages/react/src/components/RadioGroup/RadioGroupItem.tsx [hunk 1]
```diff
@@ -133,29 +132,39 @@ class RadioGroupItem extends AutoControlledComponent<
...
     if (checked !== prevState.checked) {
-      checked && this.props.shouldFocus && this.elementRef.focus()
+      checked && this.props.shouldFocus && this.elementRef.current.focus()
       _.invoke(this.props, 'checkedChanged', undefined, { ...this.props, checked })
     }
   }
 
-  componentDidMount() {
-    this.elementRef = ReactDOM.findDOMNode(this) as HTMLElement
+  handleFocus = (e: React.SyntheticEvent) => {
```
Comment (mnajdova) [inline packages/react/src/components/RadioGroup/RadioGroupItem.tsx]: 👍 for adding the handlers

File: packages/react/src/components/RadioGroup/RadioGroupItem.tsx [hunk 2]
```diff
@@ -164,25 +173,15 @@ class RadioGroupItem extends AutoControlledComponent<
               styles: styles.icon,
             },
           })}
-          {rtlTextContainer.createFor({ element: label })}
-        </Label>
-      </ElementType>
+          {Box.create(label, {
```
Comment (mnajdova) [inline packages/react/src/components/RadioGroup/RadioGroupItem.tsx:176]: Should we change the type of the `label` prop to `ShorthandValue`?

File: CHANGELOG.md [hunk 3]
```diff
@@ -16,6 +16,7 @@
 This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+- Use regular components instead of `Label` in `RadioGroupItem` @layershifter ([#XXX](https://github.com/stardust-ui/react/pull/XXX))
```
Comment (mnajdova) [inline CHANGELOG.md]: Please add section above this (Breaking change probably if we introduce changes of the typings)

**Review by kolaps33 — APPROVED** [review]
Tested with readers. And it looks good :) 👍

**Review by mnajdova — APPROVED** [review]
The code looks good. Be sure to fix the screener regressions before merging.

**Pull request merged**